core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -764,7 +778,7 @@ public TableMetadata buildReplacement(Schema updatedSchema, PartitionSpec update
       schemasBuilder.add(new Schema(freshSchemaId, freshSchema.columns(), freshSchema.identifierFieldIds()));
     }
 
-    return new TableMetadata(null, formatVersion, uuid, newLocation,
+    return new TableMetadata(null, newFormatVersion, uuid, newLocation,
```

Comment:
I think that this also needs to call `upgradeToFormatVersion` to ensure that there is no version rollback.